pandas/core/groupby/grouper.py

```diff
@@ -947,9 +948,11 @@ def is_in_obj(gpr) -> bool:
             # For the CoW case, we need an equality check as the identity check
             # no longer works (each Series from column access is a new object)
             try:
-                return gpr.equals(obj[gpr.name])
+                obj_gpr_column = obj[gpr.name]
             except (AttributeError, KeyError, IndexError, InvalidIndexError):
                 return False
+            ref = weakref.ref(obj_gpr_column._mgr.blocks[0])
+            return ref in gpr._mgr.blocks[0].refs.referenced_blocks
```

Comment:
I haven't checked yet, but I think the phenomenon in #50805 could mean `obj[grp.name]` might return a DataFrame.

Reply:
Ah - no - I think this is okay. This should always return a Series (we check provided groupers are 1d).